### ISSUE TITLE
QIcon with custom svg path produces solid filled icon

### ISSUE THREAD

**Issue by p1pchenk0** [quasarframework/quasar#6976]
**Describe the bug**
Current implementation of QIcon's render function for svgs takes user defined path and renders it in path `d` attribute:
```javascript
if (this.type.svg === true) {
      data.attrs.focusable = 'false' /* needed for IE11 */
      data.attrs.viewBox = this.type.viewBox

      return h('svg', data, mergeSlot([
        h('path', {
          attrs: { d: this.type.path }
        })
      ], this, 'default'))
    }
```
problem rises, when svg contains path with some negative parts inside of it (take a look at fiddle)
**Codepen/jsFiddle/Codesandbox (required)**
https://jsfiddle.net/m2jfsa6e/

**Expected behavior**
Svg icon is rendered correctly

**Additional context**
Problem is that svg path is rendered without `fill-rule="evenodd"` attribute.
Solution is to modify render function with one line:
```javascript
if (this.type.svg === true) {
      data.attrs.focusable = 'false' /* needed for IE11 */
      data.attrs.viewBox = this.type.viewBox

      return h('svg', data, mergeSlot([
        h('path', {
          attrs: { d: this.type.path, 'fill-rule': 'evenodd' } // fixes problem
        })
      ], this, 'default'))
    }
```

**Comment by p1pchenk0** [comment]
or to provide custom rules for path as it is done for custom view box, e.g.:
`M9 3L5 6.99h3V14h2V6.99h3L9 3zm7 14.01V10h-2v7.01h-3L15 21l4-3.99h-3z|0 0 104 104|fill-rule=evenodd`
and then use those rules for `path` element

**Comment by rstoenescu** [comment]
Hi,

Yes. Marking this as a feature request.
Will add support for some custom things, because the root cause is same as with https://github.com/quasarframework/quasar/issues/6925

**Comment by rstoenescu** [comment]
Thinking of an enhanced syntax for the svg: `path#style&path#style&...|viewBox` --> can have multiple paths, each with its own style.

For an svg with a single path and custom style, this will be: `path#style|viewBox`.

**Comment by rstoenescu** [comment]
This will allow for non-breaking change as `path|viewBox` will still be valid.

**Comment by rstoenescu** [comment]
Not sure about the delimiters though, as '#' and '&' can be valid style characters.

**Comment by p1pchenk0** [comment]
@rstoenescu so, for example above it will be `M9 3L5 6.99h3V14h2V6...z#fill-rule=evenodd|viewBox`, right? or `M9 3L5...z#{fill-rule: evenodd}|viewBox`

**Comment by rstoenescu** [comment]
Yep, u got the idea. Searching for the best delimiter chars though. Any ideas?

**Comment by p1pchenk0** [comment]
curly braces are convenient, as they are not used in path, so we can just write custom styles:
`path{style object}|viewBox`

**Comment by rstoenescu** [comment]
Final new format:

```
New format:
    
    path@@style_attr@@transform_attr&&path@@style_attr@@transform_attr&&...|viewBox
    
     - A path is defined by: path@@style_attr@@transform_attr
         * style_attr is optional
         * transform_attr is optional
         * if no style_attr but you have a transform, still use the separator between path and style_attr
           (path@@@@transform_attr)
     - Supports multiple paths, separated by "&&"
     - viewBox is optional
     - Backward compatibility with the old format (path|viewBox)
```

Will be available in "quasar" v1.11.0
